torch/_dynamo/variables/higher_order_ops.py

```diff
@@ -134,10 +134,13 @@ def validate_args_and_maybe_create_graph_inputs(
                 tracer.create_graph_input(a.as_proxy().node.name)
             new_arg = a
         else:
-            raise unimplemented(
-                f"HigherOrderOperator with body that accepts non-Tensors as input. "
-                f"Got: {a.python_type()}"
-            )
+            if manually_set_subgraph_inputs:
+                raise unimplemented(
+                    f"HigherOrderOperator with body that accepts non-Tensors as input. "
+                    f"Got: {a.python_type()}"
+                )
+            else:
+                new_arg = a
```

Comment:
Can we add tests for:
- SymInt inputs should be lifted to arguments
- non-Tensor and non-SymInt inputs (e.g. String) should not get lifted to being inputs. Constants should not be lifted to being inputs (instead they should be inlined directly into the graph, and hopefully they get guarded on). We should graph break on things that are not constants, SymInt, or Tensor.

e.g. wrap(f, (x, "foo", ["bar"])), wrap(f, (x, [SomeUserDefinedClass()]))